src/main/java/org/jenkinsci/test/acceptance/junit/WithSystemProperties.java

```diff
@@ -0,0 +1,98 @@
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * 
+ * Indicates that the test and Jenkins instance will run with the specified
+ * system properties. These properties are set as an array of pairs <i>property=vale</i>.
+ * 
+ * <p>
+ * The system properties can be set only when the test are being run with a {@link WinstoneController}.
+ * This annotation is a way to extend JENKINS_JAVA_OPTS and JENKINS_OPTS at test level
+ */
+@Retention(RUNTIME)
+@Target({METHOD, TYPE})
+@Inherited
+@Documented
+@RuleAnnotation(value = WithSystemProperties.RuleImpl.class, priority = -10)
+public @interface WithSystemProperties {
+
+    String[] value();
+
+    public class RuleImpl implements TestRule {
+        @Inject
+        JenkinsController controller;
+
+        @Override
+        public Statement apply(final Statement base, final Description d) {
+            return new Statement() {
+
+                @Override
+                public void evaluate() throws Throwable {
+                    if (!(controller instanceof WinstoneController)) {
+                        throw new AssumptionViolatedException("Test skipped. WithSystemProperties should be used with a local controller.");
+                    }
+
+                    Class<?> testSuite = d.getTestClass();
+                    final String[] fromClass = getProperties(d.getAnnotation(WithSystemProperties.class));
+                    if (fromClass != null) {
+                        for (String property : fromClass) {
+                            ((WinstoneController) controller).addSystemProperty(property);
+                        }
+                    }
+                    final String[] fromMethod = getProperties(testSuite.getAnnotation(WithSystemProperties.class));
+                    if (fromMethod != null) {
+                        for (String property : fromMethod) {
+                            ((WinstoneController) controller).addSystemProperty(property);
+                        }
+                    }
+
+                    base.evaluate();
+                }
+            };
+        }
+
+        // Visible for testing
+        static String[] getProperties(WithSystemProperties withSystemProperties) {
+            if (withSystemProperties == null) {
+                return null;
+            }
+
+            String[] properties = withSystemProperties.value();
+            if (properties == null || properties.length == 0) {
+                return null;
+            }
+
+            List<String> systemProperties = new ArrayList<>();
+            for (String property : properties) {
+                if (StringUtils.isNotBlank(property)) {
+                    if (!property.startsWith("-D")) {
+                        property = "-D" + property;
+                    }
+                    systemProperties.add(property);
+                }
+            }
+            return systemProperties.toArray(new String[0]);
```

Comment:
this is duplicated in many places.  I don't beleive you should even accept this.  assume the property is always `key=value` and use it.  then just blindly always add the `-D` when consuming it.  (or make this actually take a `key` and a `value` array rather than a string.)